docs/2025/puzzles/day06.md

```diff
@@ -2,10 +2,245 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 # Day 6: Trash Compactor
 
+by [@scarf005](https://github.com/scarf005)
+
 ## Puzzle description
 
 https://adventofcode.com/2025/day/6
 
+## Solution Summary
+
+Processing data row-by-row is usually straightforward, but handling columns can be tricky.
+Here comes [transpose](https://www.scala-lang.org/api/3.7.4/scala/collection/IterableOps.html#transpose-5d3) to the rescue!
+The `transpose` method switches the rows and columns of a 2D collection, which is exactly what we need for this puzzle.
+
+## Part 1
+
+```
+123 328  51 64 
+ 45 64  387 23 
+  6 98  215 314
+*   +   *   +
+```
+
+To make working with columns easier, we'd like the inputs to be like this:
+
+```
+123  45   6 *
+328  64  98 +
+ 51 387 215 *
+ 64  23 314 +
+```
+
+First, let's split the input into a 2D grid:
+
+```scala
+def part1(input: String) = input.linesIterator.toVector // split input into lines
+  .map(_.trim.split(raw"\s+")) // split lines into words by whitespaces
+
+// Vector(
+//   Array(123, 328, 51, 64),
+//   Array(45, 64, 387, 23),
+//   Array(6, 98, 215, 314), 
+//   Array(*, +, *, +)
+// )
+```
+
+After we transpose it, we get the desired output:
+
+```scala
+def part1(input: String) = input.linesIterator.toVector
+  .map(_.trim.split(raw"\s+"))
+  .transpose
+
+// Vector(
+//   Vector(123, 45, 6, *),
+//   Vector(328, 64, 98, +),
+//   Vector(51, 387, 215, *),
+//   Vector(64, 23, 314, +)
+// )
+```
+
+Now it's a matter of processing each column, which is fairly straightforward.
+Let's define an [extension method](https://docs.scala-lang.org/scala3/reference/contextual/extension-methods.html) to improve readability.
+
+```scala
+extension (xs: IterableOnce[(symbol: String, nums: IterableOnce[String])])
```

Comment:
If this takes an `xs: Iterator[...]` it'll be more apparent how it connects, plus...